xwiki-rendering-macros/xwiki-rendering-macro-footnotes/src/main/java/org/xwiki/rendering/internal/macro/footnote/PutFootnotesMacro.java

```diff
@@ -175,9 +175,8 @@ private ListItemBlock processFootnote(MacroMarkerBlock footnoteMacro, int counte
     private void addFootnoteRef(MacroMarkerBlock footnoteMacro, Block footnoteRef)
     {
         for (ListIterator<Block> it = footnoteMacro.getChildren().listIterator(); it.hasNext();) {
```

Comment:
Seems to be this whole code is equivalent to `footnoteMacro.getChildren().clear();`